ssl/ssl_lib.cc

```diff
@@ -2221,6 +2221,10 @@ int SSL_CTX_set_ciphersuites(SSL_CTX *ctx, const char *str) {
                                 true /* only configure TLSv1.3 ciphers */);
 }
 
+int SSL_set_ciphersuites(SSL *ssl, const char *str) {
+  return SSL_CTX_set_ciphersuites(ssl->ctx.get(), str);
```

Comment:
This seems a bit strange since I was under the impression that `SSL` objects are created from `SSL_CTX`s, so it doesn't seem quite right for `SSL_set_ciphersuites` to be configuring contents from the parent object.

[Our `SSL_CONFIG`](https://github.com/aws/aws-lc/blob/34d5e8ffd8df2424c7ef0f86108fca2c6869d349/ssl/internal.h#L3164-L3170) is sort of like OpenSSL's `SSL_CONNECTION` and contents about the `cipher_list` are saved there as well. I think we should configure the `cipher_list` in `SSL_CONFIG` instead, since it's more specific to the `SSL` structure. [`SSL(_CTX)_set_cipher_list`](https://github.com/aws/aws-lc/blob/34d5e8ffd8df2424c7ef0f86108fca2c6869d349/ssl/ssl_lib.cc#L2188-L2214) and `SSL(_CTX)_set_strict_cipher_list` may serve as good examples.

Reply:
Ah, good catch. This should be just like `SSL_set_cipher_list` and `SSL_set_strict_cipher_list` but for TLSv1.3 and not strict. Will fix.